### ISSUE TITLE
Disallow build warnings

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/runtime#6144]
The coreclr build currently generates a ridiculous number of warnings, especially on Windows where the tests are built.

Warnings should be fixed and warnings converted to build breaks to prevent them from being re-introduced.  The honor system obviously isn't cutting it.

**Comment by ViktorHofer** [comment]
@jkoritzinsky @trylek do you know if this issue is still valid?

**Comment by jkoritzinsky** [comment]
Yes this is still valid. We still issue a few warnings on Win x86 so we don’t have warn-as-error turned on.
